detox/src/artifacts/instruments/android/AndroidInstrumentsPlugin.test.js

```diff
@@ -0,0 +1,100 @@
+const AndroidInstrumentsPlugin = require('./AndroidInstrumentsPlugin');
+
+describe('AndroidInstrumentsPlugin', () => {
+  const testSummary = 'SomeTestSummary';
+  let mockedApi, mockedClient, mockedDevicePathBuilder;
+
+  beforeEach(() => {
+    mockedApi = {
+      trackArtifact: jest.fn(),
+      untrackArtifact: jest.fn()
+    };
+    mockedClient = {
+      isConnected: true,
+      pandingAppCrash: false
+    };
+    mockedDevicePathBuilder = {
+      buildTemporaryArtifactPath: () => {
+        return 'test123';
+      }
+    };
+  });
+
+  it('should not build launchArgs without started test', async () => {
+    const plugin = new AndroidInstrumentsPlugin({
+      api: {
+        ...mockedApi,
+        userConfig: {
+          enabled: true
+        }
+      },
+      client: mockedClient,
+      devicePathBuilder: mockedDevicePathBuilder
+    });
+
+    const event = {
+      launchArgs: {}
+    };
+    await plugin.onBeforeLaunchApp(event);
+    expect(event.launchArgs).toEqual({});
+  });
+
+  it('should build launchArgs with started test and empty config', async () => {
+    const plugin = new AndroidInstrumentsPlugin({
+      api: {
+        ...mockedApi,
+        userConfig: {
+          enabled: true
+        }
+      },
+      client: mockedClient,
+      devicePathBuilder: mockedDevicePathBuilder
+    });
+
+    const event = {
+      launchArgs: {}
+    };
+    await plugin.onTestStart(testSummary);
+    await plugin.onBeforeLaunchApp(event);
+    expect(event.launchArgs.detoxInstrumRecPath).toBe(mockedDevicePathBuilder.buildTemporaryArtifactPath());
+    expect(event.launchArgs.detoxInstrumSamplingInterval).toBeUndefined();
+  });
+
+  it('should build launchArgs with started test and samplingInterval from config', async () => {
+    const plugin = new AndroidInstrumentsPlugin({
+      api: {
```

Comment:
This is an optional TODO, but the preparation in every test looks really long. 
To shorten the tests, I would suggest initializing `pluginConfig` variable in `beforeEach`:

```js
pluginConfig = {
  api: mockedApi,
  client: mockedClient,
  devicePathBuilder: mockedDevicePathBuilder,
  userConfig: { enabled: true },
};
```

Then, in test preparations, you will be able to explicitly state what is different in the config that you are testing:

```js
pluginConfig.userConfig.samplingInterval = 100500;
```

instead of many lines like:

```js
   const plugin = new AndroidInstrumentsPlugin({
      api: {
        ...mockedApi,
        userConfig: {
          enabled: true
        }
      },
      client: mockedClient,
      devicePathBuilder: mockedDevicePathBuilder
    });
```